docs/aws/audit/esmonitoring/rules/es_encrypted_with_cmks.mdx

```diff
@@ -23,6 +23,199 @@ HIPAA
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Elasticsearch Service dashboard. 
+3. In the navigation pane, choose "Domains".
+4. For each domain listed in the Domains table, check the Encryption column. If the column displays "Disabled", the selected Amazon Elasticsearch domain is not encrypted with KMS CMKs.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access the Elasticsearch domains.
+
+2. Use the following AWS CLI command to list all the Elasticsearch domains:
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all Elasticsearch domain names.
+
+3. For each domain, use the following command to describe the Elasticsearch domain configuration:
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name <your-domain-name>
+   ```
+   Replace `<your-domain-name>` with the name of your Elasticsearch domain. This command will return the configuration details of the specified Elasticsearch domain.
+
+4. Check the output of the above command for the `EncryptionAtRestOptions` field. If the `Enabled` field under `EncryptionAtRestOptions` is `false`, or if the `KmsKeyId` field is not present or empty, then the Elasticsearch domain is not encrypted with KMS CMKs.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+
+2. Create a session using your AWS credentials. You can configure your credentials in the AWS credentials file. 
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create an Elasticsearch client:
+
+```python
+es_client = session.client('es')
+```
+
+4. List all Elasticsearch domains and check if they are encrypted with KMS CMKs:
+
+```python
+def check_es_encryption():
+    domains = es_client.list_domain_names()
+    for domain in domains['DomainNames']:
+        domain_name = domain['DomainName']
+        domain_info = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+        domain_encryption = domain_info['DomainStatus']['EncryptionAtRestOptions']['Enabled']
+        kms_key_id = domain_info['DomainStatus']['EncryptionAtRestOptions']['KmsKeyId']
+        if not domain_encryption or not kms_key_id:
+            print(f"Elasticsearch domain {domain_name} is not encrypted with KMS CMKs")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>